### PR TITLE
Expose the PGP key download link in the packaging details view

### PR DESCRIPTION
**Pull request by hellais**
Fixes: https://github.com/pypa/warehouse/issues/3810

### CODE REVIEW

**Comment by ewdurbin** [comment]
Before:

screen shot 2018-07-18 at 10 50 10 am


After:

screen shot 2018-07-18 at 10 49 31 am

File: warehouse/templates/packaging/detail.html [hunk 1]
```diff
@@ -246,6 +246,11 @@ <h2 class="page-title">Download files</h2>
                     <a href="{{ request.route_url('packaging.file', path=file.path) }}">
                       {{ file.filename }}
                     </a>
+                    {% if file.has_signature %}
```
Comment (ewdurbin) [inline warehouse/templates/packaging/detail.html]: Probably move this after the file size?

File: warehouse/templates/packaging/detail.html [hunk 2]
```diff
@@ -246,6 +246,11 @@ <h2 class="page-title">Download files</h2>
...
+                    <a href="{{ request.route_url('packaging.file', path=file.pgp_path) }}">
+                      <i class="fa fa-key" aria-hidden="true"></i>
```
Comment (ewdurbin) [inline warehouse/templates/packaging/detail.html]: I'd recommend using `fa-file-signature` rather than the key.

Comment (ewdurbin) [inline warehouse/templates/packaging/detail.html]: Might want to put `File Signature` to describe what this is. Just an icon is likely not sufficient.

**Comment by ewdurbin** [comment]
With suggested edits:

screen shot 2018-07-18 at 11 17 12 am

**Comment by mschwager** [comment]
This is awesome! Thanks @hellais!

**Comment by dstufft** [comment]
Hi!

Thanks for making this PR and contributing to Warehouse. However, for the reasons I expanded upon in https://github.com/pypa/warehouse/issues/3810#issuecomment-405975460, I'm going to be closing this without merging.

**Pull request closed without merging**